src/devices/machine/i8291a.h

```diff
@@ -0,0 +1,415 @@
+
+	// register r/w functions
+
+	DECLARE_WRITE8_MEMBER(dout_w);
+	DECLARE_WRITE8_MEMBER(ie1_w);
+	DECLARE_WRITE8_MEMBER(ie2_w);
+	DECLARE_WRITE8_MEMBER(spoll_mode_w);
+	DECLARE_WRITE8_MEMBER(addr_mode_w);
+	DECLARE_WRITE8_MEMBER(aux_mode_w);
+	DECLARE_WRITE8_MEMBER(addr01_w);
+	DECLARE_WRITE8_MEMBER(eos_w);
+
+	DECLARE_READ8_MEMBER(din_r);
+	DECLARE_READ8_MEMBER(ints1_r);
+	DECLARE_READ8_MEMBER(ints2_r);
+	DECLARE_READ8_MEMBER(spoll_stat_r);
+	DECLARE_READ8_MEMBER(addr_stat_r);
+	DECLARE_READ8_MEMBER(cpt_r);
+	DECLARE_READ8_MEMBER(addr0_r);
+	DECLARE_READ8_MEMBER(addr1_r);
+	void map(address_map &map);
+
+private:
+	// device-level overrides
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	void update_int(void);
+	void handle_command(void);
+	void run_fsm();
+	void run_sh_fsm();
+	void run_ah_fsm();
+	void run_dt_fsm();
+	void run_dc_fsm();
+	void run_l_fsm();
+	void run_lp_fsm();
+	void run_t_fsm();
+	void run_tp_fsm();
+	void run_tsp_fsm();
+	void run_sp_fsm();
+	void run_pp_fsm();
+	void run_rl_fsm();
+
+	// registers
+
+	uint8_t m_din;
+	uint8_t m_dout;
+	uint8_t m_ints1;
+	uint8_t m_ints2;
+	uint8_t m_ie1;
+	uint8_t m_ie2;
+	uint8_t m_address0;
+	uint8_t m_address1;
+	uint8_t m_eos;
+	uint8_t m_spoll_mode;
+	uint8_t m_address_mode;
+	uint8_t m_address_status;
+	uint8_t m_cpt;
+
+	uint8_t m_auxa;
+	uint8_t m_auxb;
+
+	bool m_atn;
+	bool m_ren;
+	bool m_nrfd;
+	bool m_ndac;
+	bool m_dav;
+	bool m_srq;
+	bool m_ifc;
+	bool m_eoi;
+	uint8_t m_dio;
+
+	bool m_nrfd_out;
+	bool m_ndac_out;
+	bool m_dav_out;
+	bool m_srq_out;
+	bool m_eoi_out;
+
+	// internal signals
+
+	bool m_pon;
+	bool m_rdy;
+	bool m_lpe;
+	bool m_ist;
+	bool m_rtl;
+	bool m_apt_flag;
+	bool m_cpt_flag;
+	bool m_din_flag;
+	bool m_nba;
+
+	bool m_pp_sense;
+	bool m_pp_line;
+
+	bool m_send_eoi;
+
+	static constexpr int REG_INTS1_BI = (1 << 0);
+	static constexpr int REG_INTS1_BO = (1 << 1);
+	static constexpr int REG_INTS1_ERR = (1 << 2);
+	static constexpr int REG_INTS1_DEC = (1 << 3);
+	static constexpr int REG_INTS1_END = (1 << 4);
+	static constexpr int REG_INTS1_GET = (1 << 5);
+	static constexpr int REG_INTS1_APT = (1 << 6);
+	static constexpr int REG_INTS1_CPT = (1 << 7);
+
+	static constexpr int REG_INTS2_ADSC = (1 << 0);
+	static constexpr int REG_INTS2_REMC = (1 << 1);
+	static constexpr int REG_INTS2_LLOC = (1 << 2);
+	static constexpr int REG_INTS2_SPC = (1 << 3);
+	static constexpr int REG_INTS2_REM = (1 << 4);
+	static constexpr int REG_INTS2_LLO = (1 << 5);
+	static constexpr int REG_INTS2_SPAS = (1 << 6);
+	static constexpr int REG_INTS2_INT = (1 << 7);
+
+	static constexpr int REG_IE2_DMAI = (1 << 4);
+	static constexpr int REG_IE2_DMAO = (1 << 5);
+
+	static constexpr int REG_ADDRESS01_ARS = (1 << 7);
+	static constexpr int REG_ADDRESS0_INT = (1 << 7);
+	static constexpr int REG_ADDRESS_DT = (1 << 6);
+	static constexpr int REG_ADDRESS_DL = (1 << 5);
+
+	static constexpr int REG_ADDRESS_STATUS_MJMN = 1 << 0;
```

Comment:
Why are you using no parentheses for these, when you're using parentheses for the ones above?